r/R/dplyr-funcs-datetime.R

```diff
@@ -412,9 +412,10 @@ register_bindings_duration <- function() {
 
     if (call_binding("is.character", x)) {
       x <- build_expr("strptime", x, options = list(format = format, unit = 0L))
-      # complex casting only due to cast type restrictions: time64 -> int64 -> duration(us)
+      # complex casting due to cast type restrictions: time64 -> int64 -> duration(us)
       # and then we cast to duration ("s") at the end
-      x <- x$cast(time64("us"))$cast(int64())$cast(duration("us"))
+      # we also need the casting chain to get the measurement units right
+      x <- make_duration(x$cast(time64("us")), unit = "us")
```

Comment:
Would you mind updating this comment to match what we have now? Since we've encapsulated the `make_duration()` complexities we don't have to worry about those. Though we might still want to keep that we're  going to `s` later on.